### ISSUE TITLE
Other test improvements from JavaScript

### ISSUE THREAD

**Issue by happy5214** [hed-standard/hed-python#10]
These are test improvements from the JavaScript code that are not in the other issues.

All of these tests are in `test_tag_validator.py` unless otherwise noted.

- There are two copies of `test_get_parent_tag_name()` in `test_hed_dictionary.py`.
- Throughout `test_hed_string_delimiter.py`, the actual values of the parsed strings should be checked instead of just their types.
- Add more tests for missing required tags to `test_check_for_required_tags()`, as in [this JS code](https://github.com/VisLab/HEDBidsValid/blob/379fb125028baf3cecfc49989308581d432c646c/tests/hed.spec.js#L698).
- Remove leaf extension tests, per #5.
- Rename `test_tag_is_valid()` and `TagValidator.tag_is_valid()` to something more descriptive (the JS code [renames](https://github.com/VisLab/HEDBidsValid/blob/379fb125028baf3cecfc49989308581d432c646c/utils/hed.js#L66) `tag_is_valid()` to `tag_exists_in_schema()`.

**Comment by owinterbg** [comment]
- Removed a copy of test_get_parent_tag_name()
- added a skip to test_check_for_required_tags() due to the linked javascript test being implemented in test_translation_hed.py
- Ignored removing leaf extension tests due to a separate issue being open
- added a skip to test_tag_is_valid() due to a better version of this test existing in test_translation_hed.py (test_exist_in_schema())
- did not rename function until Dr. Robbins gives the okay to do so

Updated to do:
- Throughout test_hed_string_delimiter.py, the actual values of the parsed strings should be checked instead of just their types.
- Rename TagValidator.tag_is_valid() to something more descriptive (the JS code renames tag_is_valid() to tag_exists_in_schema()).

**Comment by VisLab** [comment]
Can be renamed.  Please go ahead....

**Comment by owinterbg** [comment]
- renamed tag_is_valid() to tag_exist_in_schema()

Updated to do:
- Throughout test_hed_string_delimiter.py, the actual values of the parsed strings should be checked instead of just their types.

**Comment by owinterbg** [comment]
waiting on Alexander to rewrite stringparser.spec.js before completely rewriting test_hed_string_delimiter.py

**Comment by happy5214** [comment]
The JS tests have been rewritten.

**Comment by happy5214** [comment]
`test_tag_validator.py` was deprecated, rendering those fixes moot, and the remaining unfixed points of this issue (the `HedStringDelimiter` tests) have been subsumed by #67. As such, this issue will be closed.
